enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,472 @@
+
+### Risks and Mitigations
+
+#### Dead fields
+
+If an installed Gateway API CRD includes some field that is not implemented by
+the installed Gateway API implementation, then updates to this field may be
+silently ignored.  In the context of a security feature, ignoring this field
+could result in a configuration that inadvertently exposes workload.
+
+Potential mitigation strategies:
+
+- Handle with an admission webhook.
+- Work upstream to address this issue.
+- Defer the issue until we allow some newer CRD version than we initially allow.
+
+_TBD: Fill in more details._
+
+### Drawbacks
+
+> The idea is to find the best form of an argument why this enhancement should
+> _not_ be implemented.
+> 
+> What trade-offs (technical/efficiency cost, user experience, flexibility,
+> supportability, etc) must be made in order to implement this? What are the reasons
+> we might not want to undertake this proposal, and how do we overcome them?
+> 
+> Does this proposal implement a behavior that's new/unique/novel? Is it poorly
+> aligned with existing user expectations?  Will it be a significant maintenance
+> burden?  Is it likely to be superceded by something else in the near future?
+
+_TBD_
+
+## Open Questions [optional]
+
+> This is where to call out areas of the design that require closure before deciding
+> to implement the design.  For instance,
+>  > 1. This requires exposing previously private resources which contain sensitive
+>   information.  Can we do this?
+
+_TBD_
+
+## Test Plan
+
+> **Note:** *Section not required until targeted at a release.*
+> 
+> Consider the following in developing a test plan for this enhancement:
+> - Will there be e2e and integration tests, in addition to unit tests?
+> - How will it be tested in isolation vs with other components?
+> - What additional testing is necessary to support managed OpenShift service-based offerings?
+> 
+> No need to outline all of the test cases, just the general strategy. Anything
+> that would count as tricky in the implementation and anything particularly
+> challenging to test should be called out.
+> 
+> All code is expected to have adequate tests (eventually with coverage
+> expectations).
+
+The Ingress Operator will have E2E tests to simulate the user stories outlined
+in this EP:
+
+- Delete the CRDs; verify that the operator re-installs them.
+- Install incompatible CRDs with `metadata.managedFields` set to indicate that the operator *did not* install them; verify that the operator reports the appropriate `Degraded` status.
+- Install older CRDs with `metadata.managedFields` set to indicate that an older version of the operator *did* install them; verify that the operator updates them.
+
+_TBD_
+
+## Graduation Criteria
+
+Note that Gateway API CRD life-cycle management will be part of the
+[gateway-api-with-cluster-ingress-operator](gateway-api-with-cluster-ingress-operator.md) enhancement, which defines its
+own graduation criteria.
```

Comment:
We had a conversation before about potentially packaging the CRDs in at the CVO level, we should capture somewhere in this doc (alternatives considered, etc) the reasons why we moved away from that for posterity.

Reply:
I added a heading in <https://github.com/openshift/enhancements/compare/99aae59bb1ac50f1ef02e07ff3dcc3385216c451..e34441f0d0685989e6417db70dcf049c68f9fc55>.  I don't actually understand how using CVO would help, so I haven't written anything out yet.  I can go back through Jira comments and meeting notes to try to figure that out, or else feel free to write something up about that alternative.